presto-main/src/main/java/io/prestosql/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -649,9 +649,9 @@ public void abort()
     private static ListenableFuture<?> whenAllStages(Collection<SqlStageExecution> stages, Predicate<StageState> predicate)
     {
         checkArgument(!stages.isEmpty(), "stages is empty");
-        Set<StageId> stageIds = newConcurrentHashSet(stages.stream()
-                .map(SqlStageExecution::getStageId)
-                .collect(toSet()));
+        Set<StageId> stageIds = stages.stream()
+                .map(SqlStageExecution::getStageId).distinct()
```

Comment:
`.distinct()` is redundant (and would go on separate line, if retained)

(if you want to keep it, please explain why)

Reply:
not needed indeed.